gammapy/data/counts.py

```diff
@@ -0,0 +1,61 @@
+from __future__ import print_function, division
+from gammapy.spectrum import EnergyBinning
+
+
+__all__ = ['CountsSpectrumDataset',
+           ]
+
+
+class CountsSpectrumDataset(object):
+    """
+    Counts spectrum dataset (PHA wrapper).
+    
+    Test case for EnergyBinning class
+    """
+  
+    def __init__(self, counts, energy):
+
+        #does not work (why?) -> Kind of crucial
```

Comment:
Now sure what you mean.
There are a bunch of places where we do input validation, that's true:
https://github.com/gammapy/gammapy/search?utf8=%E2%9C%93&q=isinstance

I'm also not completely against it ... if you want to add it I'll still merge ... I tried to write up a loose guideline here a while back:
https://gammapy.readthedocs.org/en/latest/development/index.html#what-checks-and-conversions-should-i-do-for-inputs

Recently I saw this:
http://astropy.readthedocs.org/en/latest/api/astropy.units.quantity_input.html
Haven't tried it, but could be useful to get some input validation with little boilerplate code?